src/Internal/Payment/Request.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Class Request
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+/**
+ * Class for loading, sanitizing, and escaping data from payment requests.
+ */
+class Request {
+	/**
+	 * Request data.
+	 *
+	 * @var array
+	 */
+	private $request;
+
+	/**
+	 * The request array.
+	 *
+	 * @param  array|null $request  Request data, this can be $_POST, or WP_REST_Request::get_params().
+	 */
+	public function __construct( array $request = null ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$this->request = $request ?? $_POST;
+	}
+
+	/**
+	 * Get the fraud prevention token from the request.
+	 *
+	 * @return string|null
+	 */
+	public function get_fraud_prevention_token(): ?string {
+		return isset( $this->request['wcpay-fraud-prevention-token'] )
+			? sanitize_text_field( $this->request['wcpay-fraud-prevention-token'] )
+			: null;
+	}
+
+	/**
+	 * Check if the request is a WooPay preflight check.
+	 *
+	 * @return bool
+	 */
+	public function is_woopay_preflight_check(): bool {
+		return ! empty( $this->request['is-woopay-preflight-check'] );
```

Comment:
I think the presence of the key alone is enough for the preflight check. There is an `isset` needed anyway.

Reply:
Good point! I used `empty` because this is how the same parameter is inferred in `payment_process`. Addressed here 13e49c3a06e685406cacba965ab3df1ccac42631